lib/api/core/plugins/pluginsManager.js

```diff
@@ -313,60 +313,78 @@ function pm2Init (workers, pluginsWorker, isDummy) {
 }
 
 function initPipes (pipes, plugin, pipeWarnTime, pipeTimeout) {
-  if (plugin.config && plugin.config.pipeWarnTime !== undefined) {
-    pipeWarnTime = plugin.config.pipeWarnTime;
-  }
-  if (plugin.config && plugin.config.pipeTimeout !== undefined) {
-    pipeTimeout = plugin.config.pipeTimeout;
-  }
+  var register = (event, fn) => {
```

Comment:
Generally, I'm not really comfortable with function creation inside other functions. It's just a personal point of view, but can we create instead a private function inside the same file? Something like:
`registerPipes`
`registerHooks`

I find it difficult to read and to understand. We can discuss if you're not agree.